dependency_cache.go

```diff
@@ -454,16 +454,52 @@ func (DependencyCache) verify(path string, expected string) error {
 func (d DependencyCache) setDependencyMirror(urlD *url.URL, mirror string) {
 	if mirror != "" {
 		d.Logger.Bodyf("%s Download URIs will be overridden.", color.GreenString("Dependency mirror found."))
-		urlOverride, err := url.ParseRequestURI(mirror)
+		mirrorArgs := parseMirror(mirror)
+		urlOverride, err := url.ParseRequestURI(mirrorArgs["mirror"])
 
 		if strings.ToLower(urlOverride.Scheme) == "https" || strings.ToLower(urlOverride.Scheme) == "file" {
 			urlD.Scheme = urlOverride.Scheme
 			urlD.User = urlOverride.User
-			urlD.Path = strings.Replace(urlOverride.Path, "{originalHost}", urlD.Hostname(), 1) + urlD.Path
+			urlD.Path = strings.Replace(urlOverride.Path, "{originalHost}", urlD.Hostname(), 1) + strings.Replace(urlD.Path, mirrorArgs["skip-path"], "", 1)
 			urlD.Host = urlOverride.Host
 		} else {
 			d.Logger.Debugf("Dependency mirror URI is invalid: %s\n%w", mirror, err)
 			d.Logger.Bodyf("%s is ignored. Have you used one of the supported schemes https:// or file://?", color.YellowString("Invalid dependency mirror"))
 		}
 	}
 }
+
+// Parses a raw mirror string into a map of arguments.
+func parseMirror(mirror string) map[string]string {
+
+	mirrorArgs := map[string]string{
+		"mirror":    mirror,
+		"skip-path": "",
+	}
+
+	// Split mirror string at commas and extract specified arguments.
+	for _, arg := range strings.SplitN(mirror, ",", -1) {
```

Comment:
```suggestion
	for _, arg := range strings.Split(mirror, ",") {
```

It's probably simpler to just split (I had to look at the docs to see what -1 means). It is the same behavior as this.

> It is equivalent to [SplitN](https://pkg.go.dev/strings#SplitN) with a count of -1.

Unless we want to put a limit on things. You could probably `SplitN(mirror, ",", 2)` if we wanted to lock this down a bit more. You should really only have at most one `,` cause we only support one comma separated argument at this time and you either have that argument or you have nothing.

Reply:
The simpler split function might be more handy, compared to limiting the number of arguments. Should we have additional ones in the future, we don't need to touch this part of the code again.

I'd consider the "mirror=" key for the URL as optional and added a test case to cover this format.